chain/net/net.go

```diff
@@ -0,0 +1,176 @@
+func NextNet(reader *fileio.EasyReader, currTName string, currLvl [][]int, mp map[string]chromInfo.ChromInfo) (Net, string, [][]int, bool) {
+	line, done := fileio.EasyNextRealLine(reader)
+	if done {
+		return Net{}, currTName, currLvl, true
+	}
+	n, currTName, currLvl := NewNet(line, currTName, currLvl, mp)
+	return n, currTName, currLvl, false
+}
+
+// NewNet is a helper function for the Read function. It takes in a single line of the file representing a Net entry
+// and returns a filled in Net struct and the current chromosome
+func NewNet(text string, currTName string, currLvl [][]int, mp map[string]chromInfo.ChromInfo) (Net, string, [][]int) {
+	data := strings.Split(text, " ")
+	if data[0] == "net" {
+		currTName = data[1]
+		mp[data[1]] = chromInfo.ChromInfo{Name: data[1], Size: parse.StringToInt(data[2])}
+		return Net{}, currTName, [][]int{{0, 0}, {0}}
+	}
+	currLvl, newData := determineLevel(data, currLvl)
+	curr := Net{
+		TName:       currTName,
+		Level:       currLvl[0][0],
+		Class:       newData[0],
+		TStart:      parse.StringToInt(newData[1]),
+		TSize:       parse.StringToInt(newData[2]),
+		QName:       newData[3],
+		Orientation: parse.StringToStrand(newData[4]),
+		QStart:      parse.StringToInt(newData[5]),
+		QSize:       parse.StringToInt(newData[6]),
+		ExtraFields: strings.Join(newData[7:], " "),
+		spaces:      currLvl[0][1],
+	}
+
+	return curr, currTName, currLvl
+}
+
+// determineLevel is a helper struct for the NewNet function. It parses how many spaces proceed the data on the line
+// of the file to determine the Level of the net. It also returns the data without the proceeding spaces for easy parsing
+func determineLevel(data []string, currLvl [][]int) ([][]int, []string) {
+	var spaces int
+	for i := range data {
+		if data[i] == "" {
+			spaces++
+		} else {
+			break
+		}
+	}
+	data = data[spaces:]
+
+	if spaces == currLvl[0][1] {
+		return currLvl, data
+	} else if spaces > currLvl[0][1] {
+		if data[0] == "fill" {
+			currLvl[0][0]++
+		}
+		currLvl[1] = append(currLvl[1], currLvl[0][0])
+		currLvl[0][1] = spaces
+		return currLvl, data
+	} else if spaces < currLvl[0][1] {
+		currLvl[0][1] = spaces
+		if data[0] == "fill" {
+			currLvl[0][0] = currLvl[1][spaces]
+		} else if data[0] == "gap" {
+			currLvl[0][0] = currLvl[1][spaces-1]
+		}
+		currLvl[1] = currLvl[1][:spaces]
+		return currLvl, data
+	}
+
+	return [][]int{}, []string{}
+}
+
+// Write takes in a file name, a slice of Net and a map of chromInfo and writes out a properly formatted net file.
+func Write(outfile string, nets []Net, chromSizes map[string]chromInfo.ChromInfo) {
+	var currChrom, prevChrom string
+	file := fileio.EasyCreate(outfile)
+	for i := range nets {
+		currChrom = nets[i].TName
+		if currChrom != prevChrom {
+			fileio.WriteToFileHandle(file, fmt.Sprintf("net %s %d", currChrom, chromSizes[currChrom].Size))
+		}
+		fileio.WriteToFileHandle(file, ToString(nets[i], true))
+		prevChrom = currChrom
+	}
+	err := file.Close()
```

Comment:
I'd just wrap the close statement in the exception error handling helper function so you dont need to declare a new err variable to only be used once.

```suggestion
	exception.PanicOnErr(file.Close())
```